guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -536,7 +536,7 @@ class View extends Template
 
 6.4.1.1. Service contract interfaces SHOULD be placed in separate API modules. The other modules will depend on the API module, and implementations could be easily swapped via `di.xml`. API module namess must end with the `Api` suffix. For example, if a module is named `MyModule`, its APIs SHOULD be declared in a module named `MyModuleApi`.
 
-6.4.1.2. Service interfaces that should be exposed as web APIs MUST be placed under the `MyModuleApi/Api` directory. Service data interfaces MUST be placed under `MyModuleApi/Api/Data`. Directories under `MyModuleApi/Api` SHOULD NOT be nested.
+6.4.1.2. Service interfaces that should be exposed as web APIs MUST be placed under the `MyModuleApi/Api` directory. Service data interfaces MUST be placed under `MyModuleApi/Api/Data`. Directories under `MyModuleApi/Api` SHOULD NOT be nested. If contracts need to be added to support a module that already contains classes under `Api/` (e.g. `MyModule/Api`), the contracts SHOULD be added to the existing module.  
```

Comment:
I propose to add a note to 6.4.1.1. instead as that is the item describing the rules on a module level. So I'd rather rephrase that one as "6.4.1.1. Service contract interfaces SHOULD be placed in separate API modules **, except when existing module already contains Service Contracts in `Api` folder**. The other modules will depend on the API module, and implementations could be easily swapped via `di.xml`. API module namess must end with the `Api` suffix. For example, if a module is named `MyModule`, its APIs SHOULD be declared in a module named `MyModuleApi`."

An important point here is that `Api` folder should contain **Service Contracts** (probably that's the only expected, but who knows).